spec/tbd_tests_spec.rb

```diff
@@ -209,80 +227,100 @@
+    expect(os_g_S1_wall.uFactor).to_not be_empty
+    expect(os_g_S1_wall.uFactor.get).to be_within(TOL).of(0.31)
 
     os_v  = OpenStudio::Point3dVector.new
-    os_v << OpenStudio::Point3d.new( 46.4, 29.8, 46.0) #  2.0m
-    os_v << OpenStudio::Point3d.new( 46.4, 29.8, 44.0) #  1.0m
-    os_v << OpenStudio::Point3d.new( 47.4, 29.8, 44.0) #  2.0m
-    os_v << OpenStudio::Point3d.new( 47.4, 29.8, 46.0) #  1.0m
-
-    os_g_S_door = OpenStudio::Model::SubSurface.new(os_v, model)
-    os_g_S_door.setName("g_S_door")
-    expect(os_g_S_door.setSubSurfaceType("GlassDoor")).to be true
-    expect(os_g_S_door.setSurface(os_g_S_wall)).to be true # 2.0m2
-    expect(os_g_S_door.uFactor).to_not be_empty
-    expect(os_g_S_door.uFactor.get).to be_within(TOL).of(2.00)
+    os_v << OpenStudio::Point3d.new( 24.0, 29.8, 49.5)
+    os_v << OpenStudio::Point3d.new( 24.0, 29.8, 46.7)
+    os_v << OpenStudio::Point3d.new( 28.0, 29.8, 46.7)
+    os_v << OpenStudio::Point3d.new( 28.0, 29.8, 49.5)
+
+    os_g_S2_wall = OpenStudio::Model::Surface.new(os_v, model)
+    os_g_S2_wall.setName("g_S2_wall")
+    expect(os_g_S2_wall.setSpace(os_g)).to be true
+    expect(os_g_S2_wall.uFactor).to_not be_empty
+    expect(os_g_S2_wall.uFactor.get).to be_within(TOL).of(0.31)
 
     os_v  = OpenStudio::Point3dVector.new
-    os_v << OpenStudio::Point3d.new( 17.4, 40.2, 49.5) # 10.4m
-    os_v << OpenStudio::Point3d.new( 17.4, 29.8, 49.5) # 36.6m
-    os_v << OpenStudio::Point3d.new( 54.0, 29.8, 49.5) # 10.4m
-    os_v << OpenStudio::Point3d.new( 54.0, 40.2, 49.5) # 36.6m
+    os_v << OpenStudio::Point3d.new( 28.0, 29.8, 49.5)
+    os_v << OpenStudio::Point3d.new( 28.0, 29.8, 44.0)
+    os_v << OpenStudio::Point3d.new( 54.0, 29.8, 44.0)
+    os_v << OpenStudio::Point3d.new( 54.0, 29.8, 49.5)
+
+    os_g_S3_wall = OpenStudio::Model::Surface.new(os_v, model)
+    os_g_S3_wall.setName("g_S3_wall")
+    expect(os_g_S3_wall.setSpace(os_g)).to be true
+    expect(os_g_S3_wall.uFactor).to_not be_empty
+    expect(os_g_S3_wall.uFactor.get).to be_within(TOL).of(0.31)
+
+    os_v  = OpenStudio::Point3dVector.new
+    os_v << OpenStudio::Point3d.new( 46.4, 29.8, 46.0)
+    os_v << OpenStudio::Point3d.new( 46.4, 29.8, 44.0)
+    os_v << OpenStudio::Point3d.new( 47.4, 29.8, 44.0)
+    os_v << OpenStudio::Point3d.new( 47.4, 29.8, 46.0)
+
+    os_g_S3_door = OpenStudio::Model::SubSurface.new(os_v, model)
+    os_g_S3_door.setName("g_S3_door")
+    expect(os_g_S3_door.setSubSurfaceType("GlassDoor")).to be true
+    expect(os_g_S3_door.setSurface(os_g_S3_wall)).to be true
+    expect(os_g_S3_door.uFactor).to_not be_empty
+    expect(os_g_S3_door.uFactor.get).to be_within(TOL).of(2.00)
+
+    os_v  = OpenStudio::Point3dVector.new
+    os_v << OpenStudio::Point3d.new( 17.4, 40.2, 49.5)
+    os_v << OpenStudio::Point3d.new( 17.4, 29.8, 49.5)
+    os_v << OpenStudio::Point3d.new( 54.0, 29.8, 49.5)
+    os_v << OpenStudio::Point3d.new( 54.0, 40.2, 49.5)
```

Comment:
Quite a few changes to _Lo Scrigno_ model (the very first RSpec written to support TBD), largely to fix errors when forward translating and running EnergyPlus, including _inter alia_:

- missing constructions
- convexity issues (mainly linked to shadow casting)
- skylight area == base surface (roof) area (SDK doesn't mind, but EnergyPlus most definitely does)

One can now run EnergyPlus (from the .osm). No results to talk of, as zones don't have thermostats, loads, HVAC, etc. But there shouldn't be any geometry/construction-related errors or warnings from EnergyPlus.